lib/ansible/modules/system/mdadm.py

```diff
@@ -0,0 +1,369 @@
+
+        if force:
+            force_flag = '--force'
+
+        md_cmd = "mdadm --zero-superblock %s %s" % (disk, force_flag)
+
+        (md_rc, md_out, md_err) = self.module.run_command(md_cmd, check_rc=True)
+
+        if 'Unrecognised md component device' in md_err:
+            self.module.exit_json(changed=False, msg="Disk MD is not a MD device" %
+                                  (disk), disk=disk, error=md_err, stdout=md_out)
+            return
+
+    def scan_raid_disks(self):
+        ''' Scan existing raid disks on a system '''
+        md_cmd = "mdadm --detail --scan"
+
+        (md_rc, md_out, md_err) = self.module.run_command(md_cmd, check_rc=False)
+
+        return filter(None, md_out.split('\n'))
+
+def main():
+    argument_spec = dict(
+        state=dict(required=False, default='present',
+                   choices=['present', 'absent', 'examine', 'scan']),
+        level=dict(required=False, default=1, choices=[0, 1, 5, 6, 10], type='int'),
+        md_device=dict(required=False, type='str'),
+        spare_devices=dict(required=False, default=0, type='int'),
+        raid_devices=dict(required=False, default=None, type='list'),
+        assume_clean=dict(required=False, type='bool'),
+        force=dict(required=False, type='bool')
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+    )
+
+    state = module.params.get('state')
+    level = module.params.get('level')
+
+    md_device = module.params.get('md_device')
+    spare_devices = module.params.get('spare_devices')
+    raid_devices = module.params.get('raid_devices')
+
+    assume_clean = module.params.get('assume_clean')
+    force = module.params.get('force')
+
+    md = MdManager(module, md_device, level, raid_devices, spare_devices)
+
+    if state == 'examine':
+        if md_device:
+            module.exit_json(changed=False, **md.get_md_info(md_device))
+        elif raid_devices:
+            disk_list = []
+            for disk in raid_devices:
+                disk_list.append(md.get_raid_info(disk))
+            module.exit_json(changed=False, **dict(raid_devices=disk_list))
+    elif state == 'present':
+        if not force:
+            md.test_raid_disks(level, md_device)
+        md.test_raid_level_disks_number()
+        md.create_md_raid(force, assume_clean)
+        module.exit_json(changed=True, **md.get_md_info(md_device))
+    elif state == 'absent':
+        if md_device:
+            md.stop_md_raid(force)
+        for disk in raid_devices:
+            md.zero_md_raid(disk, force)
+        module.exit_json(changed=True,
+                         msg="Raid device '%s' was removed and wiped from disks '%s'" %
+                         (md_device, ' '.join(raid_devices)),
+                         md_device=md_device, raid_devices=raid_devices)
+    elif state == 'scan':
+        module.exit_json(changed=False, md_devices=md.scan_raid_disks())
+
+# import module snippets
+from ansible.module_utils.basic import *
```

Comment:
Nowadays we list all the functionality we use from a specific library explicitly.
So:
```python
from ansible.module_utils.basic import AnsibleModule
```